src/components/Editor/index.js

```diff
@@ -160,16 +164,35 @@ class Editor extends Component {
     } else {
       this.undoTimestamp = 0
     }
+
+    if (evt.keyCode === 27 && !this.state.tabGuarded) {// Esc Key
+      this.setState({tabGuarded:true})
+    }else if(!(evt.shiftKey || evt.keyCode === 27 || evt.keyCode === 9) && this.state.tabGuarded){
```

Comment:
Sooooo much prettier.  That's a much better way to conceptualize it. I couldn't see the forest for the trees.   It reads more like English and that always feels nice. Gets us out of that keycode "magic number" hell 🔥👿🔥   

Abstractions are a developer's best friend.